internal/builders/bazel/build.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -eu
+#
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+mkdir binaries
+
+# Transfer flags and targets to their respective arrays
+IFS=' ' read -r -a BUILD_FLAGS <<< "${FLAGS}"
+IFS=' ' read -r -a BUILD_TARGETS <<< "${TARGETS}"
+
+# Build with respect to entire arrays of flags and targets
+bazel build "${BUILD_FLAGS[@]}" "${BUILD_TARGETS[@]}"
+
+# Using target string, copy artifact to binaries dir
+for CURR_TARGET in "${BUILD_TARGETS[@]}"; do
+  # Take out the first two // in CURR_TARGET
+  # "//src/internal:fib" --> "src/internal:fib"
+  CD_PATH=$(echo "$CURR_TARGET" | cut -d'/' -f3-)
```

Comment:
If we require the targets be referenced from the workspace root this might fail (or even succeed) in some unexpected ways if users don't do that. For example, if there is a `a/b/c:main` target and a `abc:main` target this would turn `a/b/c:main` into `abc:main` and upload a different target than was intended.

Maybe we should validate it and throw an error if the target doesn't begin with `//`? Maybe also validate that the target id is specified? e.g. check for the `:`?

Reply:
Created https://github.com/slsa-framework/slsa-github-generator/issues/2308

Reply:
I agree, validation should be added in next PR.

`a/b/c:main` --> `c:main` after `CD_PATH=$(echo "$CURR_TARGET" | cut -d'/' -f3-)` since it cuts off the first two `/`'s and keeps everything after the second `/`.

The target beginning with `//` is crucial if the path to the target is longer as we can see above. Validation for target id can be added with it to check for `:` and string after it.

Thanks for this idea!